go-controller/pkg/util/util.go

```diff
@@ -420,6 +424,27 @@ func GetActiveNetworkForNamespace(namespace string, nadLister nadlister.NetworkA
 	return nil, &UnknownActiveNetworkError{namespace: namespace}
 }
 
+func GetNetworkRole(isPrimary bool) string {
+	networkRole := types.NetworkRoleSecondary
+	if isPrimary {
+		networkRole = types.NetworkRolePrimary
+	}
+	return networkRole
+}
+
+// generateExternalIDs returns the external IDs for logical switches and logical routers
+// when it runs on a primary or secondary network. It returns an empty map
+// when on the default cluster network, for backward compatibility.
+func GenerateExternalIDsForSwitchOrRouter(netInfo NetInfo) map[string]string {
+	externalIDs := make(map[string]string)
+	if netInfo.IsSecondary() {
```

Comment:
ah yes, it's time to discuss this. I could safely add external IDs to LBs that are also on the default network because secondary networks don't use services. For routers and switches I preferred not to add external IDs when on the default network, at least at first. I didn't wanna break the existing code, but let me check now what secondary networks do with them...

Reply:
In the end we decided not to alter the LBs in the default network and only add network-related external IDs to LBs in UDNs.